### PR TITLE
sddm: 0.20.0 -> unstable-2023-12-29, add Qt6 support

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

A bunch of Wayland fixes and one important tmpfiles.d change.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
@ofborg test sddm

**Comment by K900** [comment]
@ofborg build libsForQt5.sddm qt6Packages.sddm

File: pkgs/top-level/qt6-packages.nix [hunk 1]
```diff
@@ -58,6 +58,10 @@ makeScopeWithSplicing' {
     suffix = "qt6";
   };
 
+  # Not a library, but we do want it to be built for every qt version there
+  # is, to allow users to choose the right build if needed.
```
Comment (SuperSandro2000) [inline pkgs/top-level/qt6-packages.nix:62]:
```suggestion
  # Not a library, but we do want it to be built for every qt version there is,
  # to allow users to choose the right build if needed.
```

File: pkgs/applications/display-managers/sddm/default.nix [hunk 2]
```diff
@@ -1,48 +1,32 @@
-{ mkDerivation, lib, fetchFromGitHub, fetchpatch
-, cmake, extra-cmake-modules, pkg-config, qttools
-, libxcb, libXau, pam, qtbase, qtdeclarative, qtquickcontrols2, systemd, xkeyboardconfig
+{ stdenv, lib, fetchFromGitHub
+, cmake, pkg-config, qttools
+, libxcb, libXau, pam, qtbase, wrapQtAppsHook, qtdeclarative
+, qtquickcontrols2 ? null, systemd, xkeyboardconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/sddm/default.nix:4]: Why the null? Does it not exist for qt6? Is there a nicer way to solve this?

Comment (K900) [reply pkgs/applications/display-managers/sddm/default.nix:4]: In 6, QQC2 is part of qtbase.

File: pkgs/applications/display-managers/sddm/default.nix [hunk 3]
```diff
@@ -1,48 +1,32 @@
...
 }:
-mkDerivation rec {
+let
+  isQt6 = lib.versions.major qtbase.version == "6";
+in stdenv.mkDerivation {
   pname = "sddm";
-  version = "0.20.0";
+  version = "unstable-2023-12-29";
```
Comment (wegank) [inline pkgs/applications/display-managers/sddm/default.nix:10]:
```suggestion
  version = "0.20.0-unstable-2023-12-29";
```

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

File: pkgs/applications/display-managers/sddm/default.nix (hunk 2 above)
Comment (ilya-fedin) [reply pkgs/applications/display-managers/sddm/default.nix:4]: maybe qtdeclarative? qtbase doesn't have QML stuff

Comment (K900) [reply pkgs/applications/display-managers/sddm/default.nix:4]: Yeah, brainfart.

File: pkgs/top-level/qt6-packages.nix (hunk 1 above)
Comment (K900) [reply pkgs/top-level/qt6-packages.nix:62]: Reworded a little bit.

File: pkgs/applications/display-managers/sddm/default.nix (hunk 3 above)
Comment (K900) [reply pkgs/applications/display-managers/sddm/default.nix]: Thanks, applied.

**Pull request merged**